lightning/src/ln/channelmanager.rs

```diff
@@ -13029,6 +13186,21 @@ where
 			}
 		}
 
+		// Similar to the above cases for forwarded payments, if we have any pending inbound HTLCs
+		// which haven't yet been claimed, we may be missing counterparty_node_id info and would
+		// panic if we attempted to claim them at this point.
+		for (payment_hash, payment) in claimable_payments.iter() {
+			for htlc in payment.htlcs.iter() {
+				if htlc.prev_hop.counterparty_node_id.is_none() {
+					log_error!(args.logger,
```

Comment:
Wouldn't there be value in doing more of the checks that we're doing above for forwards, like checking the monitor claimable balances and the `short_to_chan_info` map and failing read if we know we're going to panic? Wondering if there's a way to `chain` these htlcs to the `monitor.get_all_current_outbound_htlcs()` above to DRY the checks.